src/test/java/de/unijena/cheminf/fragment/fingerprint/ExampleUsageTest.java

```diff
@@ -0,0 +1,214 @@
+         * O=CCl: 1
+         * OC=1C=CC=2C=CC=CC2C1: 6
+         * BrC1=CC=CC=2C=C(C=CC12)C: 1
+         * ON=[CH3]: 2
+         * ONO: 3
+         * OC1=CC=CC=2C=CC=CC12: 4
+         * NC1=CC=CC=2C=CC=CC12: 1
+         * O=C[CH]1=CC=CC=2C=CC=CC21: 2
+         * O=CO: 8
+         * ON=C: 1
+         * O=[S](=O)O: 5
+         * C=1C=CC=2C=CC=CC2C1: 20
+         * C=1C=CC=2C=C(C=CC2C1)C: 2
+         * OC1=CC=CC=2C1=CC=CC2C: 1
+         * O=N[CH]1=CC=C(O)C=2C=CC=CC21: 1
+         * OC=1C=2C=CC=CC2C=CC1C: 1
+         * [CH2][CH]=1C=CC=2C=CC=CC2C1: 1
+         * BrC1=CC=CC=2C1=CC=CC2C: 1
+         * OC1=CC=C(O)C=2C=CC=CC12: 2
+         * C=1C=CC2=C(C1)C=CC=C2C: 1
+         * O=COC: 1
+         */
+        //Collecting fragments that appear at least 2 times
+        List<String> tmpFragmentsList = new ArrayList<>(28);
+        for (String tmpFragment : tmpFrequenciesMap.keySet()) {
+            if (tmpFrequenciesMap.get(tmpFragment) > 2) {
+                tmpFragmentsList.add(tmpFragment);
+            }
+        }
+        //Initialising fingerprinter
+        FragmentFingerprinter tmpNaphthaleneFingerprinter = new FragmentFingerprinter(tmpFragmentsList);
+        System.out.println(tmpNaphthaleneFingerprinter.getSize());
+        /*
+         * Output: 7
+         *
+         * Only 7 out of the 28 fragments appear more than 2 times and are included in the fingerprint (see above).
+         */
+        //Parsing 3-hydroxy-2-naphthoic acid, fragmenting it, and creating its fingerprint
+        String tmpCNP0437667SmilesString = "O=C(O)C1=CC=2C=CC=CC2C=C1O"; //3-hydroxy-2-naphthoic acid
+        SmilesParser tmpSmiPar = new SmilesParser(SilentChemObjectBuilder.getInstance());
+        tmpFragmenter.generateFragments(tmpSmiPar.parseSmiles(tmpCNP0437667SmilesString));
+        IAtomContainer[] tmpFragments = tmpFragmenter.getFragmentsAsContainers();
+        List<String> tmpCNP0437667Fragments = new ArrayList(10);
+        for (IAtomContainer tmpFragment : tmpFragments) {
+            tmpCNP0437667Fragments.add(tmpSmiGen.create(tmpFragment));
+        }
+        IBitFingerprint tmpCNP0437667BitFP = tmpNaphthaleneFingerprinter.getBitFingerprint(tmpCNP0437667Fragments);
+        for (int i = 0; i < tmpNaphthaleneFingerprinter.getSize(); i++) {
+            System.out.println(tmpNaphthaleneFingerprinter.getBitDefinition(i) + ": " + tmpCNP0437667BitFP.get(i));
+        }
+        /*
+         * Output:
+         * BrC1=CC=CC=2C=CC=CC12: false
+         * OC=1C=CC=2C=CC=CC2C1: true
+         * ONO: false
+         * OC1=CC=CC=2C=CC=CC12: false
+         * O=CO: true
+         * O=[S](=O)O: false
+         * C=1C=CC=2C=CC=CC2C1: false
+         *
+         * 3-hydroxy-2-naphthoic acid contains the formic acid and the naphthol fragments. It does not produce a
+         * naphthalene fragment because the hydroxy fragment is too small to be considered on its own, according to the CDK
+         * ExhaustiveFragmenter.
+         */
+    }
+}
```

Comment:
These are examples rather than actual tests. But why not turn the examples into real tests?

Reply:
See the comment above, it was not my intention here to create "real" unit tests. But @B-s123 , feel free to integrate some of it into your real test classes.